lib/manager/common.ts

```diff
@@ -173,6 +174,11 @@ export interface UpdateArtifactsResult {
   file?: { name: string; contents: string };
 }
 
+export interface UpdateDependencyConfig {
+  fileContent: string;
+  updateOptions: Upgrade;
```

Comment:
Is there anything stopping this from being `upgrade`?

Reply:
Initially, it was a hard choice for me too. I realized after analyzing the complete functionality, and it is not necessarily upgrading until conditions are met.
So, all we are passing here are a couple of data points to the function to make a decision and make changes based on those conditions.

Reply:
Internally we refer to these as “upgrades” so it’s consistent that this param is called upgrade. Unless I’m forgetting something, it’s very rare that an upgrade would not lead to a branch/PR. Sometimes there’s “no work” to do, but that only means “upgrade was already applied” and not “this is not an upgrade”